src/keyframe-effect-constructor.js

```diff
@@ -67,6 +67,9 @@
       shared.deprecated('Custom KeyframeEffect', '2015-06-22', 'Use KeyframeEffect.onsample instead.');
       this._normalizedKeyframes = effectInput;
     } else {
+      if (!Array.isArray(effectInput) && this._timing.duration === 0) {
+        effectInput = [effectInput, effectInput];
+      }
```

Comment:
At t=startTime this will give the wrong result. I'm not sure we can actually support partial keyframes at all unless we plumb them into the core.